### PR TITLE
added post req handler for api creation of collection

### PR DESCRIPTION
**Pull request by HussainKhanSherwani**
# Pull Request Template

Before submitting this PR, please make sure you have completed the following checklist. This helps ensure that our codebase remains high quality and that our features work well across all supported devices and themes.

### Code Review
- [ ] I have carefully reviewed my code and believe it is final from my side.
- [ ] I have ensured there is no hardcoded color in my changes. Colors should be defined in theme files and reused.

### Theme Compatibility
- [ ] I have tested the new feature or updates in both light and dark themes.
- [ ] I confirm that the visual aspects of the feature are compatible and appealing in both themes.

### Device Compatibility
- [ ] I have tested my changes on mobile screen sizes.
- [ ] I have tested my changes on tablet screen sizes.
- [ ] I have tested my changes on laptop screen sizes.

### Functional Testing
- [ ] I have thoroughly tested the functionality of my changes.
- [ ] I am confident that my changes do not negatively impact existing features.

### Testing Documentation
Please provide a brief description of the tests you have performed to verify your changes. Include any relevant scenarios or edge cases.

- Example test 1: Tested feature X by uploading the image
- Example test 2: Tested feature X by doing these things.

# Demo Video
Please provide a demo video of your changes. This helps reviewers understand the changes and verify the functionality.

You can record the video using tools like Loom(https://www.loom.com/), Clip by ClickUp(https://clickup.com/features/clips), Sendspark(https://www.sendspark.com/), etc.

Include the link to the video below and make sure the video covers
- [ ] Explain the task/bug you are solving
- [ ] Show the changes in the UI
- [ ] Show the changes in the codebase
- [ ] Show the changes you tested to make sure it works
- [ ] Show the changes in various screen sizes
- [ ] Show the changes in both light and dark themes

### CODE REVIEW

File: academy-ui/src/app/api/[spaceId]/byte-collections/route.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  const args: CreateByteCollectionWithApiRequest = await req.json();
+  console.log(args);
+  const { spaceId } = params;
+  const apiKey = req.headers.get('X-API-KEY');
+  if (apiKey) {
+    await validateApiKey(apiKey, spaceId!);
+  }
+  const byteCollection = await prisma.byteCollection.create({
+    data: {
+      id: v4(),
```
Comment (RobinNagpal) [inline academy-ui/src/app/api/[spaceId]/byte-collections/route.ts:19]: slugify and other random v4 stuff appended to it?

File: academy-ui/src/app/api/[spaceId]/clickable-demos/[demoId]/route.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        title: args.input.title,
+        excerpt: args.input.excerpt,
+        updatedAt: new Date(),
+        steps: emptyClickableDemoModel.steps,
+      },
+    });
+  } else {
+    checkSpaceIdAndSpaceInEntityAreSame(spaceId, spaceId);
+    await checkEditSpacePermission(spaceById, req);
+    clickableDemo = await prisma.clickableDemos.upsert({
```
Comment (RobinNagpal) [inline academy-ui/src/app/api/[spaceId]/clickable-demos/[demoId]/route.ts:80]: `clickableDemo = await prisma.clickableDemos.upsert({` logic is same in the if else condition, so it should be out of if else block.

File: academy-ui/src/app/api/[spaceId]/byte-collections/route.ts (hunk 1 above)
Comment (HussainKhanSherwani) [reply academy-ui/src/app/api/[spaceId]/byte-collections/route.ts:19]: for byte collection we were not using slugify already we can add it for the new approach if you say @RobinNagpal

File: academy-ui/src/app/api/[spaceId]/byte-collections/route.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  const args: CreateByteCollectionWithApiRequest = await req.json();
+  console.log(args);
+  const { spaceId } = params;
+  const apiKey = req.headers.get('X-API-KEY');
+  if (apiKey) {
+    await validateApiKey(apiKey, spaceId!);
+  }
+  const byteCollection = await prisma.byteCollection.create({
+    data: {
+      id: createNewEntityId(args.input.name, spaceId!),
```
Comment (RobinNagpal) [inline academy-ui/src/app/api/[spaceId]/byte-collections/route.ts:19]: We don't need `!`  with spaceId here

File: academy-ui/src/app/api/[spaceId]/byte-collections/route.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import { ByteCollectionDto } from '@/types/byteCollections/byteCollection';
+import { validateApiKey } from '@/app/api/helpers/validateApiKey';
+import { createNewEntityId } from '@dodao/web-core/utils/space/createNewEntityId';
+
+async function postHandler(req: NextRequest, { params }: { params: { spaceId: string } }): Promise<NextResponse<ByteCollectionDto>> {
+  const args: CreateByteCollectionWithApiRequest = await req.json();
+  console.log(args);
+  const { spaceId } = params;
+  const apiKey = req.headers.get('X-API-KEY');
+  if (apiKey) {
```
Comment (RobinNagpal) [inline academy-ui/src/app/api/[spaceId]/byte-collections/route.ts:14]: We need to check else, the editSpacePermission

**Pull request merged**